### ISSUE TITLE
[expo-network] type still is WIFT when closed network

### ISSUE THREAD

**Issue by alantoa** [expo/expo#18960]
### Summary

`isConnected` always is `true` and the network type still is `WIFI`  when requesting `getNetworkStateAsync` method and open closed network.

https://user-images.githubusercontent.com/37520667/188161150-00a4d72b-fc66-493d-8a6e-fbc5626009ab.mp4

### What platform(s) does this occur on?

iOS

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 12.5
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 16.16.0 - /opt/homebrew/opt/node@16/bin/node
      Yarn: 3.2.0 - /opt/homebrew/bin/yarn
      npm: 8.11.0 - /opt/homebrew/opt/node@16/bin/npm
    Managers:
      CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
      Android SDK:
        API Levels: 26, 29, 31
        Build Tools: 26.0.3, 30.0.2, 30.0.3, 31.0.0, 33.0.0
        System Images: android-26 | Google APIs Intel x86 Atom_64, android-33 | Google APIs ARM 64 v8a
    IDEs:
      Android Studio: 2021.2 AI-212.5712.43.2112.8815526
      Xcode: 13.4.1/13F100 - /usr/bin/xcodebuild
    npmGlobalPackages:
      eas-cli: 0.53.1
      expo-cli: 5.4.11
    Expo Workflow: managed

### Minimal reproducible example

https://snack.expo.dev/@alantoa/expo-network-test

**Comment by Kudo** [comment]
hi there! i verified the snack example and it works for me on Expo Go 2.25.3 and ios 15.6.1. do i have any difference than your test environment?

**Comment by alantoa** [comment]
Hey @Kudo, thanks for checking it! 
As shown in the video:
- is correct to get `CELLULAR` when I closed `WLAN` and opened `Celluar Data`,
- but is incorrect to get type: `WIFI` when I opened `Airplane Mode`.



https://user-images.githubusercontent.com/37520667/188791736-e9482516-3eb3-4e86-ad91-31c4613bd798.mp4

And FYI, this issue has been [confirmed](https://showtime-rq88331.slack.com/archives/C03035K1JP9/p1662126996310669) by your team, and @tsapeta seems to know the cause.

Thank y'all! 🙏
